packages/measured-signalfx-reporter/lib/reporters/SignalFxMetricsReporter.js

```diff
@@ -0,0 +1,222 @@
+      case MetricTypes.TIMER:
+        return this._getValuesToProcessForTimer(name, metric);
+      case MetricTypes.GAUGE:
+        return this._getValuesToProcessForGauge(name, metric);
+      case MetricTypes.COUNTER:
+        return this._getValuesToProcessForCounter(name, metric);
+      case MetricTypes.HISTOGRAM:
+        return this._getValuesToProcessForHistogram(name, metric);
+      case MetricTypes.METER:
+        this._log.warn(
+          'Meters are not implemented in the signal fx reporter as you can use a counter and reduce your DPM'
+        );
+        return [];
+      default:
+        this._log.error(`Metric Type: ${type} has not been implemented to report to signal fx`);
+        return [];
+    }
+  }
+
+  /**
+   * Maps and Filters values from a Timer to a set of metrics to report to SigFx.
+   *
+   * @param {string} name The registry name
+   * @param {Timer} timer The Timer
+   * @return {MetricValueTypeWrapper[]} Returns an array of MetricValueTypeWrapper to use to build the request
+   * @protected
+   */
+  _getValuesToProcessForTimer(name, timer) {
+    let valuesToProcess = [];
+    // only grab histogram data as Meters can be accomplished with signal fx using the count from the histogram
+    valuesToProcess = valuesToProcess.concat(this._getValuesToProcessForHistogram(name, timer._histogram));
+    return valuesToProcess;
+  }
+
+  /**
+   * Maps and Filters values from a Gauge to a set of metrics to report to SigFx.
+   *
+   * @param {string} name The registry name
+   * @param {Gauge} gauge The Gauge
+   * @return {MetricValueTypeWrapper[]} Returns an array of MetricValueTypeWrapper to use to build the request
+   * @protected
+   */
+  _getValuesToProcessForGauge(name, gauge) {
+    const valuesToProcess = [];
+    valuesToProcess.push({
+      metric: `${name}`,
+      value: gauge.toJSON(),
+      type: SIGNAL_FX_GAUGE
+    });
+    return valuesToProcess;
+  }
+
+  /**
+   * Maps and Filters values from a Counter to a set of metrics to report to SigFx.
+   *
+   * @param {string} name The registry name
+   * @param {Counter} counter The data from the measure metric object
+   * @return {MetricValueTypeWrapper[]} Returns an array of MetricValueTypeWrapper to use to build the request
+   * @protected
+   */
+  _getValuesToProcessForCounter(name, counter) {
+    const valuesToProcess = [];
+    valuesToProcess.push({
+      metric: `${name}.count`,
+      value: counter.toJSON(),
+      type: SIGNAL_FX_CUMULATIVE_COUNTER
+    });
+    return valuesToProcess;
+  }
+
+  /**
+   * Maps and Filters values from a Histogram to a set of metrics to report to SigFx.
+   *
+   * @param {string} name The registry name
+   * @param {Histogram} histogram The Histogram
+   * @return {MetricValueTypeWrapper[]} Returns an array of MetricValueTypeWrapper to use to build the request
+   * @protected
+   */
+  _getValuesToProcessForHistogram(name, histogram) {
+    // TODO add full list of histogram metrics but enable filter
```

Comment:
Yes, in the SignalFx Java library they call this the "MetricDetails" to report.

Below seems like a nice implementation to me in the meantime.